test/test.cpp

```diff
@@ -33,27 +33,28 @@
 using namespace tiny_dnn::activation;
 
 #include "test_network.h"
+// FIXME(Randl) switching previous and next lines breaks tests
```

Comment:
Mhh.. seems to me more related to memory problems... Have we runned with clang or gcc adress sanitizer flags?